src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleSmartEat.kt

```diff
@@ -0,0 +1,194 @@
+
+
+
+    private val swapBackDelay by int("SwapBackDelay", 5, 1..20)
+
+    private val preferGappleHealth by float("PreferGappleHealth", 9f, 0f..20f)
+    private val preferNotchAppleHealth by float("PreferNotchAppleHealth", 2f, 0f..20f)
+    private val preferHealthPotHealth by float("PreferHealthPotHealth", 12f, 0f..20f)
+
+    private var prefersGapples = false
+    private var prefersNotchApple = false
+    private var prefersHealthPot = false
+    private val food: Pair<Int, ItemStack>?
+        get() = Hotbar.findBestItem(0) { _, itemStack ->
+            val item = itemStack.item
+
+            if (prefersHealthPot && item == Items.POTION) {
+                val hasHealthEffect =
+                    PotionUtil.getPotionEffects(itemStack).any {
+                        it.effectType == StatusEffects.INSTANT_HEALTH
+                    }
+
+                if (hasHealthEffect)
+                    return@findBestItem 100 - preferHealthPotHealth.toInt()
+            }
+
+            val foodComp = item.foodComponent ?: return@findBestItem -1
+
+            val hasHungerEffect = foodComp.statusEffects.any { it.first.effectType == StatusEffects.HUNGER }
+            if (hasHungerEffect)
+                return@findBestItem 0
+
+            if(prefersGapples && item == Items.GOLDEN_APPLE)
+                return@findBestItem 100 - preferGappleHealth.toInt()
+
+            if (prefersNotchApple && item == Items.ENCHANTED_GOLDEN_APPLE)
+                return@findBestItem 100 - preferNotchAppleHealth.toInt()
+
+            foodComp.hunger
+        }
+
+    val tickHandler = repeatable {
+        prefersGapples = player.health <= preferGappleHealth
+        prefersNotchApple = player.health <= preferNotchAppleHealth
+        prefersHealthPot = player.health <= preferHealthPotHealth
+    }
+    private object SilentOffhand : ToggleableConfigurable(this, "SilentOffhand", true) {
+        private object RenderSlot : ToggleableConfigurable(this.module, "RenderSlot", true) {
+            private val offset by int("Offset", 26, 0..40)
+            val renderHandler = handler<OverlayRenderEvent> {
+                renderEnvironmentForGUI {
+                    val currentFood = food ?: return@renderEnvironmentForGUI
+                    val dc = DrawContext(mc, mc.bufferBuilders.entityVertexConsumers)
+                    val scaledWidth = dc.scaledWindowWidth
+                    val scaledHeight = dc.scaledWindowHeight
+                    val i: Int = scaledWidth / 2
+                    val x = i - 91 - 26 - offset
+                    val y = scaledHeight - 16 - 3
+                    dc.drawItemInSlot(mc.textRenderer, currentFood.second, x, y)
+                    dc.drawItem(currentFood.second, x, y)
+                    dc.drawGuiTexture(
+                        HOTBAR_OFFHAND_LEFT_TEXTURE, i - 91 - 29 - offset,
+                        scaledHeight - 23, 29, 24
+                    )
```

Comment:
Extract those magic numbers (like 26, 91 or 3) to constants (like `XYZ_PADDING`, `ZX_MARGIN`) so a reader understands what those numbers represent

Reply:
I see where you are coming from, but I don't know if that really is the optimal solution. I tried to simply replicate Minecraft code as exactly as possible, meaning that the only reason the code should be changed is if minecraft would change their code. At this point, the easiest way of updating Liquidbounce's code would be to check in Minecraft's code again and see what has changed. Here it would just be confusing Liquidbounce's version of the code would use variables whilst Minecraft's wouldn't.